app/src/main/java/com/github/fribourgsdp/radio/UserProfileActivity.kt

```diff
@@ -92,6 +118,41 @@ class UserProfileActivity : AppCompatActivity(), PlaylistAdapter.OnPlaylistClick
         startActivity(intent)
     }
 
+    private fun instantiateViews(){
+        firebaseAuth = FirebaseAuth.getInstance()
+        usernameField = findViewById(R.id.username)
+        launchSpotifyButton = findViewById(R.id.launchSpotifyButton)
+        logoutButton = findViewById(R.id.logoutButton)
+        saveChangeButton = findViewById(R.id.saveUserButton)
+        usernameInitialText = findViewById(R.id.usernameInitial)
+        spotifyStatusText = findViewById(R.id.spotifyStatus)
+        googleSignInButton = findViewById(R.id.googleSignInButton)
+        playlistDisplay = findViewById(R.id.playlist_recycler_view)
+        userIcon = findViewById(R.id.userIcon)
+    }
+
+    //this is theoritecally never called
+    private fun createDefaultUser(){
+        user = User("Default")
+        db.setUser("defUserProfileActivity",user)
+        user.save(this)
+
+    }
+
+    private fun updateUser(){
+        val id = user.id
+        user = User(usernameField.text.toString())
+        user.id = id
+        val firebaseUser = firebaseAuth.currentUser
+        if(firebaseUser == null){
+            db.setUser(id,user)
+        }else{
+            db.setUser(firebaseUser.uid,user)
+        }
+
+        user.save(this)
+        usernameInitialText.setText(user.initial.uppercaseChar().toString())
```

Comment:
I am pretty sure that this will not update the user.. Since l.144 you call instantiate a new User, it overwrites the one currently on local storage, which deletes all the prior information we had about that user. Hence it is more of an overwrite than an update.

Reply:
Yes it is. I still have, to transfer the playlist from the old_user to the new. As discussed IRL, I will just set the user.name as a variable in order to be able to just change it.